### PR TITLE
Make hbase2 profile as default; switch to hbase2 in distribution #915  

### PR DESCRIPTION
**Pull request by jerryjch**
Signed-off-by: Jerry He <jerryjch@apache.org>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by jerryjch** [comment]
@FlorianHockmann Thanks for the review.
The question is if we bundle hbase2 in the distribution, hbase2 client is not garenteed to be compatible with hbase1 server. This may mean

**Comment by jerryjch** [comment]
@FlorianHockmann  Thanks for the review.
The question is that we will bundle hbase2 in JanusGraph 4.0, since hbase2 client is compatible with hbase1 server, how do we claim the versions we support in JanusGraph 4.0?

**Comment by chupman** [comment]
Hey @jerryjch and @FlorianHockmann. For `versions.adoc` we should start with `2.1.0` since that's we have in the root pom and what I assume is being used in CI. Any additional versions we would add should be dependent either on someone manually verifying or adding conditional CI tests like we did with Cassandra.

Separately, right now on the releases we claim Tested Compatibility on HBase 1.2.6, 1.3.1, 1.4.4, but I don't think 1.3.1 and 1.4.4 are actually getting run in CI so I'll open a ticket to add `[full build]` jobs for them as well. Any additional HBase 2 versions could be added in the same manner.

**Comment by jerryjch** [comment]
@FlorianHockmann @chupman 
I updated the upgrade.adoc and versions.adoc.

File: docs/upgrade.adoc [hunk 1]
```diff
@@ -7,6 +7,10 @@ JanusGraph release.
 
 === Upgrading to JanusGraph 0.4.0
 
+The version of HBase that is included in the distribution of JanusGraph was upgraded from 1.2.6 to 2.1.5.
+HBase 2.x client is not fully backward compatible with HBase 1.x server. Users who operrate their own HBase version 1.x cluster may need to upgrade their cluster to version 2.x.
```
Comment (chupman) [inline docs/upgrade.adoc]: typo, should be operate.

Comment (chupman) [inline docs/upgrade.adoc]: typo, should be operate.

Comment (jerryjch) [reply docs/upgrade.adoc]: Corrected.

File: docs/upgrade.adoc [hunk 2]
```diff
@@ -7,6 +7,10 @@ JanusGraph release.
 
 === Upgrading to JanusGraph 0.4.0
 
+The version of HBase that is included in the distribution of JanusGraph was upgraded from 1.2.6 to 2.1.5.
```
Comment (FlorianHockmann) [inline docs/upgrade.adoc]: (nitpick) Please add a headline like:
```
==== HBase: Upgrade from 1.2 to 2.1
```

All upgrade docs for this version are backend specific and users probably only want to read the aspects for their backends.

Comment (jerryjch) [reply docs/upgrade.adoc]: Done.

**Comment by goelrajat** [comment]
When will these changes be available in a release ? What is the plan for 0.4 release?

**Comment by FlorianHockmann** [comment]
> When will these changes be available in a release ? What is the plan for 0.4 release?

We can probably release 0.4.0 next week.

**Pull request merged**